src/services/pool.ts

```diff
@@ -109,9 +143,42 @@ const mapBranches = (branches: BranchEntry[], fullBaseUrl: string) => {
   }));
 };
 
-const mapRepos = (repo: RepoDocument): RepoResponse => ({
-  repoName: repo.repoName,
-  project: repo.project,
-  search: repo.search,
-  branches: mapBranches(repo.branches, getRepoUrl(repo.url[ENV_URL_KEY], repo.prefix[ENV_URL_KEY])),
-});
+const mapRepos = async (repo: RepoDocument) => {
+  try {
+    const query: Filter<DocsetDocument> = { project: repo.project };
+    const strictOptions: FindOptions = {
+      projection: {
+        project: 1,
+        url: 1,
+        prefix: 1,
+      },
+    };
+
+    const repo_docset = await db
+      .collection<DocsetDocument>(DOCSETS_COLLECTION)
+      .find(query, strictOptions)
+      .map(mapDocsets)
+      .toArray();
+
+    return {
+      repoName: repo.repoName,
+      project: repo.project,
+      search: repo.search,
+      branches: mapBranches(
+        repo.branches,
+        getRepoUrl(repo_docset[0].url[ENV_URL_KEY], repo_docset[0].prefix[ENV_URL_KEY])
+      ),
+    };
+  } catch (e) {
+    logger.error(createMessage(`Error while finding docsets: ${e}`));
+    throw e;
+  }
+};
+
+const mapDocsets = (docset: DocsetDocument) => {
+  return {
+    project: docset.project,
+    url: docset.url,
+    prefix: docset.prefix,
+  };
+};
```

Comment:
Do we need to have this function for mapping? Seems like this is doing the same thing as the projection